authorize_helper.go

```diff
@@ -22,14 +22,34 @@
 package fosite
 
 import (
+	"html/template"
+	"io"
 	"net/url"
 	"regexp"
+	"strconv"
 	"strings"
+	"time"
+
+	"golang.org/x/net/html"
+	goauth "golang.org/x/oauth2"
 
 	"github.com/asaskevich/govalidator"
 	"github.com/pkg/errors"
 )
 
+var formPostTemplate = template.Must(template.New("form_post").Parse(`<html>
+   <head>
+      <title>Submit This Form</title>
+   </head>
+   <body onload="javascript:document.forms[0].submit()">
+      <form method="post" action="{{ .RedirURL }}">
```

Comment:
`RedirURL` should be probably escaped before blindly included in the HTML template?